holoviews/plotting/bokeh/chart.py

```diff
@@ -906,18 +904,38 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            try:
+                xdiff = np.diff(xvals)
+                if len(np.unique(xdiff)) == 1 and xdiff[0] == 0:
+                    xdiff = 1
+                if is_dt:
+                    width = xdiff.astype('timedelta64[ms]').astype(np.int32) * width
```

Comment:
I think the `astype` conversion are unsafe in both cases. 

1) if you xdiff with difference off 1 ns this will give wrong values: `np.timedelta64('1', 'ns').astype('timedelta64[ms]')`
2) If the xdiff values are large, you are downgrading 64bit to 32 bit: `np.timedelta64(2**32 + 1, 'ms').astype(np.int32)`

In an ideal world this should be done by a function making it easier to work with other data-backend than numpy (like polars).

Reply:
Okay, I copied the original off of Philipp's commit. Maybe changing to `[us]` and `np.int64` addresses your concerns.

Reply:
> if you xdiff with difference off 1 ns this will give wrong values: np.timedelta64('1', 'ns').astype('timedelta64[ms]')

This will always give bad results because bokeh does not support nanosecond precision. Only difference is whether the precision is lost here or during serialization.

>If the xdiff values are large, you are downgrading 64bit to 32 bit: np.timedelta64(2**32 + 1, 'ms').astype(np.int32)

Yeah, not sure why I used int32, JS (and therefore Bokeh) supports at most integers up to 2**53 – 1.